salt/profiles/config/etc-nginx-sites-enabled-profiles.conf

```diff
@@ -1,6 +1,16 @@
 {% from 'elife/nginx-macros.sls' import consumer_groups_filter %}
 {{ consumer_groups_filter(pillar.profiles.consumer_groups_filter) }}
 
+upstream docker {
+    # 'max_fails=1' (default) will disable the upstream for 'fail_timeout=10' seconds (default)
+    # 'max_fails=0' disables the number of failed attempts to consider before upstream is marked as unavailable for 10 seconds.
+    # see:
+    # - https://nginx.org/en/docs/http/ngx_http_upstream_module.html#max_fails
+    # - https://nginx.org/en/docs/http/ngx_http_upstream_module.html#fail_timeout
+    #server localhost:9000 max_fails=0;
```

Comment:
`/etc/hosts` looks like:

```
127.0.0.1		localhost prod--profiles.elifesciences.org
# The following lines are desirable for IPv6 capable hosts
::1		ip6-localhost ip6-loopback localhost prod--profiles.elifesciences.org
...
```

with **two** mentions of `localhost` in there. Perhaps a better fix would be reverting back to 'localhost' and fixing the `hosts` file? 

hrm, no. It's set here: https://github.com/elifesciences/builder-base-formula/blob/master/elife/hostname.sls#L16-L23

I guess the problem then is that the docker container network is ipv4 only or the app is ipv4 only?

the docker container /etc/hosts looks like:

```
elife@prod--profiles:~$ docker exec -it 4ec6881dd262 bashsh
www-data@4ec6881dd262:/srv/profiles$ cat /etc/hosts
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
...
```

so the problem is in the Flask app then? I can't see it hardcoded anywhere.

Reply:
aha! ahafuckingha: https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.run

hardcoded by Flask.

todo: explain why 'localhost' can't be used in this case. Easier to change nginx setting than app code.